azerothcore/azerothcore-wotlk
# refactor(Core/Cache): Move the GlobalPlayerCache to its own class

## Changes Proposed:
-  Move the global cache to its own class
-  Fixes change faction not working due to mail cache overflow
- Implement commands to check and debug the global cache (cache info, cache delete and cache refresh)
- Cache refresh allows reseting the cache data for the given player if something wrong happens
- Delete caches will only refresh correctly the next time players relog (use it for debugging only)

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested in-game
- 


## How to Test the Changes:


1. Send mails/delete mails
2. use the .cache info $playerName command
3. Check if the count displayed matches & you can faction change normally

## Known Issues and TODO List:


- [ ] merge https://github.com/azerothcore/mod-azerothshard/pull/15/files


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
